intranet/apps/announcements/notifications.py

```diff
@@ -163,16 +163,21 @@ def announcement_posted_twitter(request, obj):
             text = "{}... - {}".format(title[:110], url)
         logger.debug("Posting tweet: %s", text)
 
-        resp = notify_twitter(text)
-        respobj = json.loads(resp)
-
-        if respobj and "id" in respobj:
-            messages.success(request, "Posted tweet: {}".format(text))
-            messages.success(request, "https://twitter.com/tjintranet/status/{}".format(respobj["id"]))
+        try:
+            resp = notify_twitter(text)
+            respobj = json.loads(resp)
+        except (ValueError, requests.RequestException, json.JSONDecodeError) as e:
+            # requests_oauthlib exceptions inherit from ValueError
+            messages.error(request, f"Error posting to Twitter: {e}")
+            logger.debug(f"Error posting to Twitter: {e.__class__}: {e}")
```

Comment:
This isn't a blocker for this merge, but, in the future, we should probably capture and send these errors to Sentry since this means that the application is suffering an error.